File: problems/2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        hashmap = {}
        for arr in words:
            for cha in arr:
                if cha in hashmap:
                    hashmap[cha] = hashmap[cha] + 1
                else:
                    hashmap[cha] =  1
    
        for key, value in hashmap.items():
            if (value % len(words) != 0):
                return False
        return True

File: problems/2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
const makeEqual = (words) => {
    if (words.length === 1) return true;
    const chaCount = {};
    for (let arr of words) {
        for (let cha of arr) {
           chaCount[cha] = (chaCount[cha] || 0) + 1; 
        }
    }
    const values = Object.values(chaCount);
    for (let val of values) {
        if (val % words.length !== 0) {
            return false;
        }
    }
    return true;
    // return values.every(count => count % words.length === 0);
};